js/animator.js

```diff
@@ -148,6 +148,60 @@ var animator = animator || {};
       this.frameWebps.push(promise);
     }
 
+    loadFromFile(file) {
```

Comment:
You should be able to reuse a lot of the logic in capture() and/or load(), rather than duplicating it here.

Reply:
Indeed; but I'm not too fluent in JS, nor the browser API; I started with a proof-of-concept, and might have a go at cleaning it up one of these days :)

Reply:
Split off common functionality into new functions.